app/components/signature-pill.js

```diff
@@ -11,84 +11,48 @@ export default class SignaturePillComponent extends Component {
   @service intl;
   @service currentSession;
 
-  @tracked signingHubUrl;
-  @tracked isMarked = false;
-  @tracked isPrepared = false;
-  @tracked hasToBeApproved = false;
-  @tracked hasToBeSigned = false;
-  @tracked isSigned = false;
-  @tracked isRefused = false;
-  @tracked isCancelled = false;
-
-  constructor() {
-    super(...arguments);
-
-    this.loadData.perform();
-  }
+  status = trackedFunction(this, async () => {
+    const signMarkingActivity = await this.args.signMarkingActivity;
```

Comment:
Then this can also be used to solve the "accessing `this` after `await`" issue.

You can do:
```
const piece = this.args.piece;
const signMarkingActivity = this.args.signMarkingActivity;
if (this.status.uri !== SIGNED) {
   ... rest of the code ...
} 
return null;
```

Reply:
Why not? Not sure from where, but there is a promise being passed as arg here

Reply:
if args is a proxy we should be awaiting it in controller/component where we use this component.
2 out 4 places where this pill is used have the data loaded before rendering the component. Should be 4 out of 4.
I would argue that if there is no `signMarkingActivity` then we should not even render the pill at all.